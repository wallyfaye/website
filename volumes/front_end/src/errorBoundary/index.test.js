import React, { Component } from 'react';
import ErrorBoundary from './index';
import renderer from 'react-test-renderer';

class ErrorComponent extends Component {
  render() {
    throw new Error('ErrorComponent throws error')
    return <div>ErrorComponent</div>
  }
}

/** @test {ErrorBoundary} */
test('ErrorBoundary should render', () => {
  const component = renderer.create(
    <ErrorBoundary>
      <ErrorComponent />
    </ErrorBoundary>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
