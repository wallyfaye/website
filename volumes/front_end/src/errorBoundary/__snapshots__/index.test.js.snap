// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render 1`] = `
<div>
  <p>
    Something went wrong.
  </p>
</div>
`;
